malleefowl/custom_workflow.py

```diff
@@ -0,0 +1,266 @@
+from owslib.wps import WebProcessingService
+from owslib.wps import ComplexDataInput
+
+from dispel4py.workflow_graph import WorkflowGraph
+from dispel4py import simple_process
+from dispel4py.core import GenericPE
+
+from malleefowl.workflow import run as run_basic_workflow
+
+import logging
+logger = logging.getLogger("PYWPS")
+
+
+class MonitorPE(GenericPE):
+    def __init__(self):
+        GenericPE.__init__(self)
+
+        self._monitor = None
+        self._pstart = 0
+        self._pend = 100
+
+    def set_monitor(self, monitor, start_progress=0, end_progress=100):
+        self._monitor = monitor
+        self._pstart = start_progress
+        self._pend = end_progress
+
+
+class GenericWPS(MonitorPE):
+    STATUS_NAME = 'status'
+    STATUS_LOCATION_NAME = 'status_location'
+
+    def __init__(self, url, identifier, inputs=[], linked_inputs=[], headers=None, **kwargs):
+        MonitorPE.__init__(self)
+
+        def log(message, progress):
+            if self._monitor:
+                self._monitor("{0}: {1}".format(
+                    self.identifier, message),
+                    progress)
+            else:
+                logger.info('STATUS ({0}: {2}/100) - {1}'.format(
```

Comment:
It is a better practise to use % for string formatting in a logging message as it is only evaluated if needed, depending on log level.